testsuite/cluster-test/src/health/mod.rs

```diff
@@ -108,6 +113,7 @@ impl HealthCheckRunner {
         events: &[ValidatorEvent],
         affected_validators_set: &HashSet<String>,
         print_failures: PrintFailures,
+        full_node_check: bool,
```

Comment:
Why do we have `full_node_check` param in run method?
I understand some setups don't support full node health check, should we just initialize HealthCheckRunner without full node health check in this case?

Reply:
I wanna save one query time especially in fn ```run_single_experiment```, and in ```wait_until_all_healthy```, which is using same HealthCheckRunner, also we do call ```HealthCheckRunner``` twice in one experiment, but we also only need one time full node check at end of the experiment. I have thought about this point, instead of add a new member of ```HealthCheckRunner``` with different initializer in ```ClusterTestRunner```, maybe a flag is much easier and flexible?

Reply:
I don't think querying prometheus is expensive and have this extra flag is a bit strange(HealthCheckRunner otherwise is not aware of health check specifics)

I think we want full node healthcheck everywhere, not just in the end of experiment, same way we have it for other health checks

I think what worries me more right now is that our prometheus seem to be flacky, but that's separate issue

Reply:
yes, prometheus is sometimes flacky and returned nothing, so this is why I added protection to go through check if prometheus is flacky.

querying prometheus one time with 30 nodes and 30 validators will return around 5Mb data(by chromn tool) and we also need to process it. I think it will not become critical with current performance test framework, but what if we have more nodes and validators? If this will be a case I think saving times of querying should be there, but if it not a problem, I can remove this flag and run full node check always

Reply:
So yes, I think we need to run this healthcheck always.

We need to address flaky prometheus as well, obviously more often we query prometheus more often test would fail if it is flaky.

To be honest, maybe given flakiness of prometheus, we can just query node for the metric using debug interface, .e.g. `AsyncNodeDebugClient::get_node_metric`?

Reply:
ok, I can make full node health check always at this stage.

I did not read debug interface code I can have a look first. Once we decide make some change of this part, we can discuss again then?